### PR TITLE
Changed find_city in DeclarationConverter and added find_country

### PR DESCRIPTION
**Pull request by EmmaSamoilova**

### CODE REVIEW

File: business_register/converter/declaration.py [hunk 1]
```diff
@@ -107,7 +108,8 @@ def save_property(self, property_data, declaration):
 
     # TODO: retrieve country from Country DB
     def find_country(self, property_country_data):
-        pass
+        country = Country.objects.get(nacp_id=property_country_data)
+        return country
```
Comment (IegorSoboliev) [inline business_register/converter/declaration.py:112]: Consider a case when property_country_data is incorrect

**Comment by roman-tiukh** [comment]
image

File: business_register/converter/declaration.py [hunk 2]
```diff
@@ -107,7 +108,14 @@ def save_property(self, property_data, declaration):
...
     def find_country(self, property_country_data):
-        pass
+        if property_country_data.isdigit():
+            country = Country.objects.filter(nacp_id=property_country_data).first()
+            if country:
+                return country
+            else:
+                logger.error(f'Cannot find country id {property_country_data}')
+        else:
+            logger.error(f'Invalid value {property_country_data}')
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:118]: need to add declaration identification, to known which declaration throw error

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

**Review by roman-tiukh — APPROVED** [review]

**Comment by roman-tiukh** [comment]
@IegorSoboliev

File: business_register/converter/declaration.py [hunk 3]
```diff
@@ -47,7 +48,7 @@ def save_unfinished_construction(self, unfinished_construction_data, declaration
     #      'percent-ownership_extendedstatus', 'seller', 'ua_middlename', 'rightBelongs', 'ua_company_code',
     #      'ua_company_name', 'ua_firstname', 'ua_livingAddressFull_extendedstatus', 'rights_id'}
     # ]
-    def save_property_right(self, property, acquisition_date, rights_data):
+    def save_property_right(self, property, acquisition_date, rights_data, declaration):
```
Comment (IegorSoboliev) [inline business_register/converter/declaration.py:51]: Declaration already is in property)

File: business_register/converter/declaration.py [hunk 4]
```diff
@@ -192,11 +193,24 @@ def save_property(self, property_data, declaration):
             person = data.get('person')
             rights_data = data.get('rights')
             if rights_data:
-                self.save_property_right(property, acquisition_date, rights_data)
+                self.save_property_right(property, acquisition_date, rights_data, declaration)
```
Comment (IegorSoboliev) [inline business_register/converter/declaration.py:196]: you can pass here only nacp_declaration_id

File: business_register/converter/declaration.py [hunk 5]
```diff
@@ -217,22 +230,23 @@ def split_address_data(self, address_data):
...
+            logger.error(f'cannot find district {district} in nacp_declaration_id {nacp_declaration_id}')
         else:
             city_of_registration = RatuCity.objects.filter(
                 name=city,
                 region=ratu_region,
                 district=ratu_district
             ).first()
-        return city_of_registration
+            print(city_of_registration)
+            return city_of_registration
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:248]: print?

Comment (EmmaSamoilova) [reply business_register/converter/declaration.py:248]: tested and forgot to remove :)

File: business_register/converter/declaration.py [hunk 6]
```diff
@@ -215,12 +220,12 @@ def save_property(self, property_data, declaration):
...
             additional_info = data.get('otherObjectType', '')
             # TODO: add country
-            # property_country = self.find_country(data['country'])
+            property_country = self.find_country(data['country'])
             property_city = None
             property_location = data.get('ua_cityType')
             # TODO: add property_city
             # if property_location:
-            #     property_city = self.find_city(property_location)
+            #     property_city = self.find_city(property_location, declaration)
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:228]: wrong comment

File: business_register/converter/declaration.py [hunk 7]
```diff
@@ -257,7 +262,16 @@ def save_property(self, property_data, declaration):
...
     def find_country(self, property_country_data):
-        pass
+        if property_country_data.isdigit():
+            country = Country.objects.filter(nacp_id=property_country_data).first()
+            if country:
+                return country
+            else:
+                self.log_error(
+                    f'Cannot find country id {property_country_data}'
+                )
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:272]:
```suggestion
                self.log_error(f'Cannot find country id {property_country_data}')
```

File: business_register/converter/declaration.py [hunk 8]
```diff
@@ -382,9 +396,7 @@ def save_declaration(self):
             )
             declarations_data = response.json().get('data')
             if response.status_code != 200 or not declarations_data:
-                logger.error(
-                    f'cannot find declarations of the PEP with nacp_declarant_id: {nacp_declarant_id}'
-                )
+                self.log_error(f'cannot find declarations of the PEP')
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:399]: here `self.current_declaration` is have wrong value or None

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: business_register/converter/declaration.py [hunk 9]
```diff
@@ -437,6 +448,6 @@ def save_declaration(self):
                 #     self.save_unfinished_construction(detailed_declaration_data['step_4']['data'], declaration)
 
                 # 'Step_5' - declarant`s family`s unfinished construction
-                if (detailed_declaration_data['step_5']
-                        and not detailed_declaration_data['step_5'].get('isNotApplicable')):
-                    self.save_luxuries(detailed_declaration_data['step_5']['data'], declaration)
+                # if (detailed_declaration_data['step_5']
+                #         and not detailed_declaration_data['step_5'].get('isNotApplicable')):
+                #     self.save_luxuries(detailed_declaration_data['step_5']['data'], declaration)
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:453]: these changes do not apply to the task

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

**Review by roman-tiukh — CHANGES_REQUESTED** [review]
have unresolved comments

**Comment by roman-tiukh** [comment]
unresolved
image

**Review by roman-tiukh — APPROVED** [review]

File: business_register/converter/declaration.py [hunk 10]
```diff
@@ -313,6 +318,53 @@ def save_property(self, property_data, declaration):
...
+        if district and not ratu_district:
+            self.log_error(f'Cannot find district {district}')
+        else:
+            city_of_registration = RatuCity.objects.filter(
+                name=city,
+                region=ratu_region,
+                district=ratu_district
+            ).first()
+            return city_of_registration
+        self.log_error(f'Cannot find city')
```
Comment (IegorSoboliev) [inline business_register/converter/declaration.py:366]: Consider adding address_data to the error`s message

**Review by IegorSoboliev — APPROVED** [review]

**Pull request merged**